src/main/kotlin/three/consulting/epoc/service/FirebaseService.kt

```diff
@@ -0,0 +1,63 @@
+package three.consulting.epoc.service
+
+import com.google.auth.oauth2.GoogleCredentials
+import com.google.firebase.FirebaseApp
+import com.google.firebase.FirebaseOptions
+import com.google.firebase.auth.FirebaseAuth
+import mu.KotlinLogging
+import org.springframework.stereotype.Service
+import three.consulting.epoc.common.Role
+import three.consulting.epoc.dto.EmployeeDTO
+import three.consulting.epoc.dto.FirebaseUserDTO
+import three.consulting.epoc.entity.Employee
+import three.consulting.epoc.repository.EmployeeRepository
+
+private val logger = KotlinLogging.logger {}
+
+@Service
+class FirebaseService(
+    private val employeeRepository: EmployeeRepository
+) {
+    fun firebaseAuth(): FirebaseAuth? {
+        try {
+            val options = FirebaseOptions.builder()
+                .setCredentials(GoogleCredentials.getApplicationDefault())
+                .build()
+            val firebaseApp = FirebaseApp.initializeApp(options)
+            return FirebaseAuth.getInstance(firebaseApp)
+        } catch (e: java.lang.Exception) {
+            logger.error(e) { "Could not set up Firebase" }
+        }
+        return null
+    }
+
+    private val firebaseAuth = firebaseAuth()
```

Comment:
Instead of defining the auth method and calling it here, why not just define it as a bean inside FirebaseConfig?
```kotlin
@Profile("default")
@Configuration
class FirebaseConfig(
    private val firebaseService: FirebaseService
) : InitializingBean {
    override fun afterPropertiesSet() {
    ...
    }

    @Bean
    fun firebaseAuth() =
        try {
            val options = FirebaseOptions.builder()
                .setCredentials(GoogleCredentials.getApplicationDefault())
                .build()
            val firebaseApp = FirebaseApp.initializeApp(options)
            return FirebaseAuth.getInstance(firebaseApp)
        } catch (e: Exception) {
            logger.error(e) { "Could not set up Firebase" }
            throw e
        }
```
This way firebaseauth is initialized in the configuration when everything is injected to spring context and it's grouped inside a configuration and not a service.

Also there's no need to use `java.lang.Exception`. Kotlin has it's own shorthands so you can just use `Exception` without importing anything